src/components/map/Layers/Countries.tsx

```diff
@@ -54,80 +89,107 @@ function FilterCountryEstimates(map: mapboxgl.Map, estimateGradePrevalences: Est
+                            });
+                        }
+                    });
+                }
+                else {
                     SetCountryEstimates(map, estimateGradePrevalences);
                 }
-            });
+            }
+            else if(countryFocus)
+            {
+                SetStudiesMode(map, countryFocus);
+            }
         }
-    }, [estimateGradePrevalences, map]);
+    }, [map, estimateGradePrevalences, countryFocus]);
 
-    // If estimates are updated, waits until map is loaded then filters country features
-    useEffect(() => {
-        if (estimateGradePrevalences.length > 0 && map) {
-            FilterCountryEstimates(map, estimateGradePrevalences)
-        }
-    }, [estimateGradePrevalences, map]);
 
     // wait until map is loaded then creates and binds popup to map events
     useEffect(() => {
         if (map && popup === undefined) {
-            const countryPop = new mapboxgl.Popup({ offset: 25, className: "pin-popup", closeButton: false });
-
-            map.on("mouseenter", "Countries", function (e: any) {
-                if (e.features[0].state.hasData) {
-                    countryPop.setMaxWidth("250px")
-                        .setHTML(ReactDOMServer.renderToString(CountryPopup(e.features[0], state.language)))
-                        .setLngLat(e.lngLat)
-                        //.trackPointer()
-                        .addTo(map);
-                }
+            const countryPop = new mapboxgl.Popup({
+              offset: 25,
+              className: "pin-popup",
+              closeOnClick: true,
+              closeButton: true,
+              anchor: 'top-left'
             });
 
-            map.on("mouseleave", "Countries", function (e: any) {
-                if (countryPop.isOpen()) {
-                    countryPop
-                    .remove()
-                }
-            });
+            setPopup(countryPop)
 
-            map.on("mousemove", "Countries", function (e: any) {
-                const f = e.features[0];
-                if (f.state.hasData && f.id !== currentFeatureId && countryPop.isOpen()) {
+            map.on('click', COUNTRY_LAYER_ID, function(e: any) {
+                
+                if (map.queryRenderedFeatures(e.point).filter((f) => f.source === "study-pins").length === 0) {
+                    const country = e.features[0];
                     countryPop
-                    .setLngLat(e.lngLat)
-                    .setHTML(ReactDOMServer.renderToString(CountryPopup(e.features[0], state.language)))
-                    setCurrentFeatureId(f.id);
-                }
+                      .setHTML(ReactDOMServer.renderToString(
+                          CountryPopup(
+                            country, 
+                            state.language, 
+                            country.state?.studyConfig ? (() => history.push(`/${state.language}/${country.state.studyConfig.viewUrl}`)) : null )))
```

Comment:
We seem to be mixing "undefined" and "null" as nullable states.

To prevent any weird errors I think we should either:
a) Decide on one of those nullable states and stick to it
b) be less specific when performing null checks (e.g. `if (map?.getSource(COUNTRY_LAYER_ID))` rather than `if (map?.getSource(COUNTRY_LAYER_ID) === undefined)`)

Option a is probably safer